app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,252 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.contexts.ContextTask;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName(value = "TaskServiceTest 에서")
+class TaskServiceTest {
+
+    @Nested
+    @DisplayName("generateId() 매소드는")
+    class Describe_generateId {
+
+        @Nested
+        @DisplayName("신규 할일이 추가되면")
+        class Context_withNewTask extends ContextTask {
+
+            final Task firstCreatedTask;
+            final Task secondCreatedTask;
+
+            public Context_withNewTask() {
+                this.firstCreatedTask = taskService.createTask(generateNewTask(TASK_TITLE_1));
+                this.secondCreatedTask = taskService.createTask(generateNewTask(TASK_TITLE_2));
+            }
+
+
+            @Test
+            @DisplayName("신규 할일의 번호를 기존 번호에 1만큼 증가한 값으로 자동생성한다.")
+            void it_automatically_generates_incremented_taskId() {
+                Long idGap = secondCreatedTask.getId() - firstCreatedTask.getId();
+                assertThat(idGap).isEqualTo(1L);
+            }
+        }
+    }
```

Comment:
테스트를 잘 작성하셨습니다! Context에 맞게 실험을 준비하는 메소드도 만들어 두셨고, `@Test`에서는 테스트에만 집중하고 있습니다. 생성자를 쓴 것도 final을 쓸 수 있으므로 괜찮은 선택입니다. 다만, 실험 준비 코드에 `@BeforeEach`를 붙여주는 관례가 있으니 다음과 같은 방법을 사용하는 것도 고려해 보세요. 양쪽을 다 할 수 있어야 합니다.

```suggestion
    @Nested
    @DisplayName("generateId() 매소드는")
    class Describe_generateId {

        @Nested
        @DisplayName("신규 할일이 추가되면")
        class Context_withNewTask extends ContextTask {

            private Task firstCreatedTask;
            private Task secondCreatedTask;

            @BeforeEach
            public void prepareTest() {
                this.firstCreatedTask = taskService.createTask(generateNewTask(TASK_TITLE_1));
                this.secondCreatedTask = taskService.createTask(generateNewTask(TASK_TITLE_2));
            }

            @Test
            @DisplayName("신규 할일의 번호를 기존 번호에 1만큼 증가한 값으로 자동생성한다.")
            void it_automatically_generates_incremented_taskId() {
                Long idGap = secondCreatedTask.getId() - firstCreatedTask.getId();
                assertThat(idGap).isEqualTo(1L);
            }
        }
    }
```